packages/synapse-react-client/src/components/GenericCard/Linkify.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import { TargetEnum } from '../CardContainerLogic'
+import { Link } from '@mui/material'
+import { PRODUCTION_ENDPOINT_CONFIG } from '../../utils/functions/getEndpoint'
+
+export type LinkifyProps = {
+  text?: string
+  className?: string
+}
+
+export type LinkifyRule = {
+  regex: RegExp
+  onMatch: (value: string) => string
+}
+
+const synapseIdRule: LinkifyRule = {
+  regex: /(syn\d+(?:\.\d+)?)/,
+  onMatch: value => `${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${value}`,
+}
+
+const httpRule: LinkifyRule = {
+  regex: /((ftp|http|https):\/\/[^ ",]+)/,
+  onMatch: value => value,
+}
+
+const pubMedRule: LinkifyRule = {
+  regex: /(PMID:\d+)/,
+  onMatch: value => {
+    const id = value.slice('PMID:'.length)
+    return `https://pubmed.ncbi.nlm.nih.gov/${id}`
+  },
```

Comment:
Could change the capture group so you don't have to `slice`, but it's up to you
```suggestion
  regex: /PMID:(\d+)/,
  onMatch: value => `https://pubmed.ncbi.nlm.nih.gov/${value}`
```

Reply:
Your way is arguably more readable, so I'm ok with either